elytron/WFLY-14984-encrypted-expression-for-client-config.adoc

```diff
@@ -0,0 +1,172 @@
+= Add support for encrypted expressions in the wildfly-config.xml
+:author:            Prarthona Paul
+:email:             prpaul@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+WildFly includes a common configuration framework that can be used to configure multiple clients, rather than relying on each client having their own configuration file. The configuration can be defined using a `wildfly-config.xml` file, where there are multiple configuration sections containing attributes whose values determine the configuration of the WildFly clients. 
+
+Attribute values for client configuration can include sensitive information. And while credential references can be used for some of them, others still need to be defined using plaintext. As a result, those attributes are still exposed and make the application vulnerable. Additionally, the password for the credential store needs to be defined as clear-text, which is not ideal. 
+
+This feature adds support for encrypted expressions for attributes in `wildfly-config.xml` file, which allows users to avoid specifying sensitive information using plaintext. Instead, encrypted expressions can be used to specify these information. A configuration section named `encrypted-expression` would be added, which would reference a `secret-key-credential-store` to extract encrypted expressions. It would also have a resolver, which would be backed by the secret-key-credential-store. The `secret-key-credential-store` can be created either using the `elytron-tool`.
```

Comment:
"to extract encrypted expressions" should be reworded. The credential store contains the secret key that is used to encrypt/decrypt encrypted expressions.

Reply:
"The `secret-key-credential-store` can be created either using the `elytron-tool`."

This sentence is incomplete I think.

Reply:
"It would also have a resolver, which would be backed by the secret-key-credential-store"

Will the resolver be able to reference a credential-store as well in addition to being able to reference a secret-key-credential-store?

Reply:
Resolvers are backed by aliases from a [CredentialStore](https://wildfly-security.github.io/wildfly-elytron/documentation/api/eap74/org/wildfly/security/credential/store/package-summary.html) object. Secret-key-credential-store is an instance of that class with type PropertiesCredentialStore. So, yes, we can use either type. 
I will add a note about that. Thanks!